src/firefly/js/visualize/ui/MatchLockDropDown.jsx

```diff
@@ -10,10 +10,40 @@ import {DropDownToolbarButton} from '../../ui/DropDownToolbarButton.jsx';
 import MATCH_LOCKED from 'html/images/28x28_Match_Locked.png';
 import MATCH_UNLOCKED from 'html/images/28x28_Match_Unlocked.png';
 import {isImage} from '../WebPlot';
+import {showInfoPopup} from 'firefly/ui/PopupUtil';
 
 function changeMatchType(vr, matchType, lockMatch) {
     const plot= primePlot(vr);
     if (!plot) return;
+    //check for images without WCS when aligning by WCS or Target, and display appropriate warning
+    if(matchType.key === 'Standard' || matchType.key === 'Target') {
+        const plotViewObjs = getPlotViewAry(vr);
+        const warningTitles = [];
+        for (const i in plotViewObjs) {
+            const currPlots = plotViewObjs[i].plots;
+            for (const j in currPlots) {
+                const plt = currPlots[j];
+                const checkWCSExists = hasWCSProjection(plt);
+                if (!checkWCSExists) {
+                    if (!warningTitles.includes(plt.title)) {
+                        warningTitles.push(plt.title);
+                    }
+                }
+            }
+        }
+
+        if (warningTitles.length !== 0) { //at least one image without WCS
+            const msgTitle = 'The following image(s) do not have WCS:';
+            const msgDesc = (String(warningTitles)).replace(/,/g, ', '); //warningTitles.join('\r\n');
```

Comment:
`warningTitles.join(', ');` does not work?

Reply:
Yes that worked, didn't think of it, way simpler. Changed it.